feature/platform/tests/optics_power_and_bias_current_test/optics_power_and_bias_current_test.go

```diff
@@ -0,0 +1,179 @@
+		expectedMaxOutPower float64
+		checkMinOutPower    bool
+	}{{
+		// Check both input and output optics power are in normal range.
+		desc:                "Check initial input and output optics powers are OK",
+		IntfStatus:          true,
+		expectedStatus:      telemetry.Interface_OperStatus_UP,
+		expectedMaxOutPower: maxOpticsPower,
+		checkMinOutPower:    true,
+	}, {
+		desc:                "Check output optics power is very small after interface is disabled",
+		IntfStatus:          false,
+		expectedStatus:      telemetry.Interface_OperStatus_DOWN,
+		expectedMaxOutPower: minOpticsPower,
+		checkMinOutPower:    false,
+	}, {
+		desc:                "Check output optics power is normal after interface is re-enabled",
+		IntfStatus:          true,
+		expectedStatus:      telemetry.Interface_OperStatus_UP,
+		expectedMaxOutPower: maxOpticsPower,
+		checkMinOutPower:    true,
+	}}
+	for _, tc := range cases {
+		t.Log(tc.desc)
+		intUpdateTime := 2 * time.Minute
+		t.Run(tc.desc, func(t *testing.T) {
+			i.Enabled = ygot.Bool(tc.IntfStatus)
+			dut.Config().Interface(dp.Name()).Replace(t, i)
+			dut.Telemetry().Interface(dp.Name()).OperStatus().Await(t, intUpdateTime, tc.expectedStatus)
+
+			// TODO: Remove t.Skipf() after the issue is fixed.
+			t.Skipf("Optics from INNOLIGHT is not supported, skip it for now.")
+
+			channels := dut.Telemetry().Component(dp.Name()).Transceiver().ChannelAny()
+			inputPowers := channels.InputPower().Instant().Get(t)
+			outputPowers := channels.OutputPower().Instant().Get(t)
+			for _, inPower := range inputPowers {
+				if inPower > maxOpticsPower || inPower < minOpticsPower {
+					t.Errorf("Get inputPower for port %q): got %.2f, want within [%f, %f]", dp.Name(), inPower, minOpticsPower, maxOpticsPower)
+				}
+			}
+			for _, outPower := range outputPowers {
+				if outPower > tc.expectedMaxOutPower {
+					t.Errorf("Get outPower for port %q): got %.2f, want < %f", dp.Name(), outPower, tc.expectedMaxOutPower)
+				}
+				if tc.checkMinOutPower && outPower < minOpticsPower {
+					t.Errorf("Get outPower for port %q): got %.2f, want > %f", dp.Name(), outPower, minOpticsPower)
+				}
+			}
+		})
+	}
+}
+
+func findComponentsByType(t *testing.T, dut *ondatra.DUTDevice, cType telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) []string {
+	components := dut.Telemetry().ComponentAny().Name().Get(t)
+	var s []string
+	for _, c := range components {
+		lookupType := dut.Telemetry().Component(c).Type().Lookup(t)
+		if !lookupType.IsPresent() {
+			t.Logf("Component %s type is not found", c)
+		} else {
+			componentType := lookupType.Val(t)
+			t.Logf("Component %s has type: %v", c, componentType)
+
+			switch v := componentType.(type) {
+			case telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT:
+				if v == cType {
+					s = append(s, c)
+				}
+			default:
+				t.Fatalf("Expected component type to be a hardware component, got (%T, %v)", componentType, componentType)
```

Comment:
Why do we fatal if there are non-hardware components? Software component is also allowed by OpenConfig. How about just ignore the component?

Reply:
done